File: common/src/main/java/net/shiroha233/roadweaver/network/ClientNetBridge.java
package net.shiroha233.roadweaver.network;

import dev.architectury.injectables.annotations.ExpectPlatform;
import net.minecraft.resources.ResourceLocation;
import net.shiroha233.roadweaver.client.map.MapLoadPhase;

/**
 * 客户端网络桥接
 */
public final class ClientNetBridge {
    private ClientNetBridge() {}

    @ExpectPlatform
    public static void requestSnapshot(int requestSeq,
                                       ResourceLocation dimensionId,
                                       MapLoadPhase phase,
                                       int responseIndex,
                                       int minX,
                                       int minZ,
                                       int maxX,
                                       int maxZ) {
        throw new AssertionError();
    }

    @ExpectPlatform
    public static void requestTeleport(int x, int y, int z) {
        throw new AssertionError();
    }

    @ExpectPlatform
    public static void requestManualConnect(int ax, int az, int bx, int bz) {
        throw new AssertionError();
    }
}

File: common/src/main/kotlin/net/shiroha233/roadweaver/network/ClientNetBridge.kt
package net.shiroha233.roadweaver.network

import dev.architectury.injectables.annotations.ExpectPlatform

object ClientNetBridge {
    @JvmStatic
    @ExpectPlatform
    fun requestSnapshot(minX: Int, minZ: Int, maxX: Int, maxZ: Int) {
        throw AssertionError()
    }

    @JvmStatic
    @ExpectPlatform
    fun requestTeleport(x: Int, y: Int, z: Int) {
        throw AssertionError()
    }

    @JvmStatic
    @ExpectPlatform
    fun requestManualConnect(ax: Int, az: Int, bx: Int, bz: Int) {
        throw AssertionError()
    }
}
